### PR TITLE
td-shim: move `event_log` out to `tcg` crate

### PR DESCRIPTION
**Pull request by gaojiaqi7**
- Move generic TCG definitions out to the `tcg` crate.
 - Keep TD-Shim related event log definitions in the `td-shim/event_log`.
 - Move the `CCEL` ACPI table definition to `td-shim/acpi`.

Signed-off-by: Jiaqi Gao <jiaqi.gao@intel.com>

### CODE REVIEW

File: tcg/src/log.rs [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    offset: usize,
+    last: usize,
+    extender: Box<dyn Fn(&[u8; SHA384_DIGEST_SIZE], u32)>,
+}
+
+impl<'a> CcEventLogWritter<'a> {
+    pub fn new(
+        area: &mut [u8],
+        extender: Box<dyn Fn(&[u8; SHA384_DIGEST_SIZE], u32)>,
+    ) -> Option<CcEventLogWritter> {
```
Comment (jodh-intel) [inline tcg/src/log.rs:27]: It might be more conventional to return a `Result<>` here?

File: tcg/src/log.rs [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    }
+
+    fn calculate_digest_and_extend(
+        &self,
+        hash_data: &[u8],
+        mr_index: u32,
+    ) -> [u8; SHA384_DIGEST_SIZE] {
+        let hash_value = ring::digest::digest(&ring::digest::SHA384, hash_data);
+        let hash_value = hash_value.as_ref();
+        assert_eq!(hash_value.len(), SHA384_DIGEST_SIZE);
```
Comment (jodh-intel) [inline tcg/src/log.rs:76]: Just querying this assert. Would it be better to protect it with `#[cfg(test)]`, or should this be a `panic!()`, or should the fucntion return a `Result<>`?

**Review by jodh-intel — COMMENTED** [review]
Thanks @gaojiaqi7 - a few comments...

File: tcg/src/log.rs (hunk 2 above)
Comment (gaojiaqi7) [reply tcg/src/log.rs:76]: It should always be `SHA384_DIGEST_SIZE` when we use `SHA384` to calculate the digest, is it good to use assert here?

**Comment by jyao1** [comment]
Based upon https://github.com/confidential-containers/td-shim/issues/405, I recommend using "cc_event_log" as crate name instead of "tcg".

**Review by jyao1 — APPROVED** [review]

**Pull request merged**